File: src/main/java/com/habit_tracker/backend/auth/service/AuthService.java
package com.habit_tracker.backend.auth.service;

import com.habit_tracker.backend.auth.dto.AuthResponse;
import com.habit_tracker.backend.auth.dto.LoginRequest;
import com.habit_tracker.backend.auth.dto.RegisterRequest;

public interface AuthService {

  AuthResponse register(RegisterRequest request);

  AuthResponse login(LoginRequest request);
}

File: mobile/HabitsTracker/HabitsTracker/auth/AuthService.swift
import Foundation

final class AuthService {
    static let shared = AuthService()

    func login(email: String, password: String) async throws -> AuthResponse {
        let body = LoginRequest(login: email, password: password)

        return try await APIClient.shared.request(
            "/auth/login",
            method: "POST",
            body: body,
            responseType: AuthResponse.self
        )
    }

    func register(name: String, email: String, password: String) async throws -> AuthResponse {
        let body = RegisterRequest(username: name, email: email, password: password)

        return try await APIClient.shared.request(
            "/auth/register",
            method: "POST",
            body: body,
            responseType: AuthResponse.self
        )
    }
}

